ramanujan/enumerators/RelativeGCFEnumerator.py

```diff
@@ -0,0 +1,283 @@
+from time import time
+from typing import List
+from collections import namedtuple
+import mpmath
+
+from ramanujan.CachedSeries import CachedSeries
+from ramanujan.constants import g_N_verify_compare_length, g_N_initial_key_length
+from .AbstractGCFEnumerator import AbstractGCFEnumerator, Match
+
+RefinedMatch = namedtuple('RefinedMatch', 'lhs_key rhs_an_poly rhs_bn_poly lhs_match_idx c_top c_bot precision')
+
+IterationMetadata = namedtuple('IterationMetadata', 'an_coef bn_coef')
+
+FIRST_STEP_MIN_ITERS = 7
+FIRST_STEP_MAX_ITERS = 100
+FIRST_STEP_BURST_NUMBER = 7
+
+SECOND_STEP_MIN_ITERS = 7
+SECOND_STEP_MAX_ITERS = 10_000
+SECOND_STEP_BURST_NUMBER = 7
+
+
+class ZeroInAn(Exception):
+    pass
+
+
+class NotConverging(Exception):
+    pass
+
+
+# TODO - pass this function to utils.
+def trunc_division(p, q):
+    """ Integer division, rounding towards zero """
+    sign = (p < 0) + (q < 0) == 1  # if exactly one is negative
+    div = abs(p) // abs(q)
+    return -div if sign else div
+
+
+def gcf_calculation_to_precision(an_iterator, bn_iterator, result_precision, min_iters, burst_number):
+    """
+    Calculate the GCF's value to a changing dept that depends on the GCF's convergence rate.
+
+    Most GCFs will converge by oscillation. If we calculate the GCF's value on the n and n+1 dept, we can conclude
+    (for most GCFs) that the resulting value will be between those two values. Using this principle, we'll stop the
+    GCF's calculation when those two values have the same decimal value under the required approximation.
+
+    If the value didn't converge and the iterators are not yielding any new items, we'll take the last two values
+    calculated for the GCF and compare their digits. As long as the digits are the match - we'll assume that it is a
+    real value.
+
+    Since huge int division is a costly process, we'll do so only every burst_number of calculations.
+
+    Returns an integer with the matching key for the GCF, which is int(gcf_value*precision_factor) and the number of
+    digits calculated.
+    """
+    computed_values = []
+    items_computed = 0
```

Comment:
Consider using a [deque](https://docs.python.org/3/library/collections.html#collections.deque) with a maxlen of 3. The only difference will be lower memory consumption.